File: app/src/main/java/org/techtown/find_gas_station/Fragment/GasolineFragment.java
package org.techtown.find_gas_station.Fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.github.mikephil.charting.charts.LineChart;

import org.techtown.find_gas_station.MVVM.GetOilViewModel;
import org.techtown.find_gas_station.R;


public class GasolineFragment extends Fragment {

    private GetOilViewModel getOilViewModel;
    private LineChart lineChart;
    private RecyclerView oilAvg_recyclerView;
    private TextView priceText;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getOilViewModel = new ViewModelProvider(this).get(GetOilViewModel.class);
        //getOilViewModel 초기화




    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View rootView =  inflater.inflate(R.layout.fragment_gasoline, container, false);

        oilAvg_recyclerView = rootView.findViewById(R.id.oilAvg_recyclerView);
        oilAvg_recyclerView.setLayoutManager(new LinearLayoutManager(requireActivity()));

        priceText = rootView.findViewById(R.id.priceText);
        lineChart = rootView.findViewById(R.id.line_chart);
        getOilViewModel.getOilAvg(lineChart,oilAvg_recyclerView,priceText,"B027");





        return rootView;
    }
}

File: app/src/main/java/org/techtown/find_gas_station/View/Fragment/GasolineFragment.kt
package org.techtown.find_gas_station.View.Fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import org.techtown.find_gas_station.databinding.FragmentGasolineBinding

class GasolineFragment : Fragment() {

    private val getOilViewModel by lazy { ViewModelProvider(this)[GetOilViewModel::class.java] }
    private lateinit var mBinding : FragmentGasolineBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        mBinding = FragmentGasolineBinding.inflate(layoutInflater, container,false)
        mBinding.oilAvgRecyclerView.layoutManager = LinearLayoutManager(requireActivity())
        getOilViewModel!!.getOilAvg(mBinding.lineChart, mBinding.oilAvgRecyclerView, mBinding.priceText, "B027")

        return mBinding.root

    }
}